### ISSUE TITLE
Dropping official Appveyor support?

### ISSUE THREAD

**Issue by joerick** [joerick/cibuildwheel#308]
Appveyor's build times have gone from bad to worse, and it really holds up the green light on PRs. In addition, I don't heed its failures anymore because they are often random and unrelated to the code. Would the community be okay dropping official support and removing it from our CI? 

Github Actions and Azure Pipelines are great alternatives, and seem very reliable.

**Comment by joerick** [comment]
I do feel bad that Appveyor gave their service to open source for free for so long, only for Microsoft to move into the space and take the market share in only a couple years. But Appveyor would have held on if their service had been better...

**Comment by Czaki** [comment]
Hm. Build time is enormous. But maybe add `--retries 3` to `pip install` may help with unexpected fails?

Also their plan is not good for building wheel because they offer only 2 concurrent job on top plan. 

Maybe other solution is to test only windows build (like in release 1.0 for support old configuration). This will reduce build time.

Did you try to contact with them?

**Comment by YannickJadoul** [comment]
Yeah, I feel the same, indeed. I've been thinking to have a look at Azure, as well, for myself, but then AppVeyor now dóes work. It's just so annoying you only have one build at a time.
So I personally still use AppVeyor, but of course, cibuildwheel changing would give me the push to invest a few hours into Azure/GitHub actions/...

But, practical consideration/suggestion: does it really take so much extra time to support AppVeyor? Could we keep AppVeyor support, but just ignore the test? Or maybe only build on master and branches that start with "appveyor" (to debug failures)?
Somehow, Windows always seems to be the annoying case, so having a diversity of Windows builds might be good for catching bugs and minor issues?

**Comment by YannickJadoul** [comment]
Or, could we skip most tests except for the basic ones (except maybe on `master`)?
Something tells me this should be reasonably easy setting a pytest environment variable before calling the tests?

**Comment by Czaki** [comment]
> But, practical consideration/suggestion: does it really take so much extra time to support AppVeyor? Could we keep AppVeyor support, but just ignore the test? Or maybe only build on master and branches that start with "appveyor" (to debug failures)?

But remove test on appveyor (which is question of this issue) does not block running cibuildwhell on appveyor.

**Comment by YannickJadoul** [comment]
> But remove test on appveyor (which is question of this issue) does not block running cibuildwhell on appveyor.

No, but if we want to not remove support for AppVeyor, we should at least have some tests running to make sure it probably works. My point is that we don't need to run all the tests all the time, and can still be fairly confident about AppVeyor being supported.

**Comment by Czaki** [comment]
If I good remember most problem is with windows, so maybe left windows test? And add everywhere `--retry 3` to prevent fail from network problem?

**Comment by joerick** [comment]
Hm, yeah, maybe we can keep some kind of 2nd-tier support, but not run the cibuildwheel test suite there. If we set appveyor to only test branches starting with `appveyor-`, then if bug reports come in with some issue specific to Appveyor we can debug it in a PR, and then merge back to master once that's done. I haven't seen many Appveyor-specific issues over the years, especially since we switched to using nuget on Windows.
